test/t8_cmesh/t8_gtest_cmesh_set_partition_offsets.cxx

```diff
@@ -0,0 +1,164 @@
+ protected:
+  void
+  SetUp () override
+  {
+    ieclass = std::get<0> (GetParam ());
+    inum_trees = std::get<1> (GetParam ());
+  }
+  t8_eclass_t ieclass;
+  t8_gloidx_t inum_trees;
+};
+
+/* call t8_cmesh_offset_concentrate for non-derived cmesh 
+ * and destroy it before commit. */
+TEST_P (cmesh_set_partition_offsets_nocommit, test_set_offsets)
+{
+  t8_cmesh_t cmesh;
+  const int main_process = 0;
+
+  t8_debugf ("Testing t8_cmesh_set_partition_offset (no commit) with %li trees.\n", inum_trees);
+
+  /* Build a valid offset array. For this test it is onlt necessary that 
+   * the array corresponds to any valid partition.
+   * We use the offset_concentrate function to build an offset array for a partition
+   * that concentrates all trees at one process. */
+  t8_shmem_init (sc_MPI_COMM_WORLD);
+  t8_shmem_array_t shmem_array = t8_cmesh_offset_concentrate (main_process, sc_MPI_COMM_WORLD, inum_trees);
+
+  /* Initialize the cmesh */
+  t8_cmesh_init (&cmesh);
+
+  /* Set the partition offsets */
+  t8_cmesh_set_partition_offsets (cmesh, shmem_array);
+  /* Destroy the cmesh */
+  t8_cmesh_unref (&cmesh);
+}
+
+/* call t8_cmesh_offset_concentrate for non-derived cmesh 
+ * and commit it. */
+TEST_P (cmesh_set_partition_offsets_commit, test_set_offsets)
+{
+  t8_cmesh_t cmesh;
+  const int main_process = 0;
+  sc_MPI_Comm comm = sc_MPI_COMM_WORLD;
+
+  t8_debugf ("Testing t8_cmesh_set_partition_offset (with commit) with %li trees of class %s.\n", inum_trees,
+             t8_eclass_to_string[ieclass]);
+
+  /* Build a valid offset array. For this test it is only necessary that 
+   * the array corresponds to any valid partition.
+   * We use the offset_concentrate function to build an offset array for a partition
+   * that concentrates all trees at one process. */
+  t8_shmem_init (comm);
+  t8_shmem_array_t shmem_array = t8_cmesh_offset_concentrate (main_process, comm, inum_trees);
+
+  /* Initialize the cmesh */
+  t8_cmesh_init (&cmesh);
+
+  /* Set the partition offsets */
+  t8_cmesh_set_partition_offsets (cmesh, shmem_array);
+
+  /* Specify a dimension */
+  const int dim = t8_eclass_to_dimension[ieclass];
+  t8_cmesh_set_dimension (cmesh, dim);
+
+  /* Set class for the trees */
+  for (int itree = 0; itree < inum_trees; ++itree) {
+    t8_cmesh_set_tree_class (cmesh, itree, ieclass);
+  }
+
+  /* Commit the cmesh */
+  t8_cmesh_commit (cmesh, comm);
+
+  /* Check that the cmesh was partitioned according to the offset */
+
+  ASSERT_TRUE (t8_cmesh_is_committed (cmesh));
+
+  /* Get the mpirank */
+  int mpirank;
+  int mpiret;
+
+  mpiret = sc_MPI_Comm_rank (comm, &mpirank);
+  SC_CHECK_MPI (mpiret);
+
+  /* Get the local number of trees */
+  const t8_locidx_t num_local_trees = t8_cmesh_get_num_local_trees (cmesh);
+  /* Compute the reference value, num_trees for mpirank main_process,
+   * 0 on each other rank. */
+  const t8_locidx_t expected_num_local_trees = mpirank == main_process ? inum_trees : 0;
```

Comment:
inum_trees is a `t8_gloidx_t`, either make `expected_num_local_trees` a `t8_gloidx_t`, or check that `inum_trees` fits in a `t8_locidx_t`

Reply:
done